tests/Integration/Behaviour/Features/Context/Domain/CarrierFeatureContext.php

```diff
@@ -100,14 +100,21 @@ public function createCarrier(string $reference, TableNode $node): void
                 $properties['logoPathName'] = DummyFileUploader::upload($properties['logoPathName']);
             }
 
+            $taxRulesGroupId = (int) TaxRulesGroupFeatureContext::getTaxRulesGroupByName($properties['taxRuleGroup'])->id;
+
             $carrierId = $this->createCarrierUsingCommand(
                 $properties['name'],
                 $properties['delay'],
                 (int) $properties['grade'],
                 $properties['trackingUrl'],
                 (int) $properties['position'],
-                (bool) $properties['active'],
-                $properties['logoPathName'] ?? null
+                filter_var($properties['active'], FILTER_VALIDATE_BOOLEAN),
```

Comment:
I don't know if you knew about it but we have a `PrimitiveUtils` class that helps ith such transformations, although it's optional because the way you used works fine as well Just so you know it has other useful helpers

Reply:
As I see, `PrimitiveUtils::castStringBooleanIntoBoolean` use bootval. But, for some cases like "yes" or "no" for instance, we don't retrieve the correct boolean. So I prefer for the behat tests to have this `filter_var` function. So, in the tests we can use 1, true, yes and  in no case-sensitive way.